R/drop_shared.R

```diff
@@ -1,26 +1,94 @@
-
-
 #' Creates and returns a shared link to a file or folder.
 #'
-#' Returns a list with the share \code{url}, \code{expires}, and \code{visibility}.
 #' @template path
-#' @template locale
-#' @param  short_url By default the function return shortened URL. Set to \code{FALSE} if you require a full path.
+#' @param requested_visibility Can be `public`, `team_only`, or `password`. If the
+#'   passsword option is chosen one must specify the `link_password`. Note that
+#'   for basic (i.e. free) Dropbox accounts, the only option is to publicly
+#'   share. Private sharing requires a pro account.
+#' @param link_password The password needed to access the document if
+#'   `request_visibility` is set to password.
+#' @param expires Set the expiry time. The timestamp format is
+#'   "\%Y-\%m-\%dT\%H:\%M:\%SZ"). If no timestamp is specified, link never expires
 #' @template token
 #' @export
 #' @examples \dontrun{
 #' write.csv(mtcars, file = "mt.csv")
 #' drop_upload("mt.csv")
 #' drop_share("mt.csv")
+#' # If you have a pro account, you can share files privately
+#' drop_share("mt.csv", requested_visibility = "password", link_password = "test")
 #'}
 drop_share <- function(path = NULL,
-                       locale = NULL,
-                       short_url = TRUE,
+                       requested_visibility = "public",
+                       link_password = NULL,
+                       expires = NULL,
                        dtoken = get_dropbox_token()) {
-   args <- as.list(drop_compact(c(path = path,
-                                locale = locale,
-                                short_url = short_url)))
-   share_url <- "https://api.dropbox.com/1/shares/auto/"
-   res <- GET(share_url, query = args, config(token = dtoken))
-   pretty_lists(content(res))
+  # This is a list because in the POST call it becomes a nested JSON.
+  # A sample response looks like this:
+
+  # curl -X POST https://api.dropboxapi.com/2/sharing/create_shared_link_with_settings \
+  #  --header "Authorization: Bearer <get access token>" \
+  #  --header "Content-Type: application/json" \
+  #  --data "{\"path\": \"/Prime_Numbers.txt\",\"settings\": {\"requested_visibility\": \"public\"}
+
+  # Check to see if only supported modes are specified
+  visibilities <- c("public", "team_only", "password")
+  assertive::assert_any_are_matching_fixed(visibilities, requested_visibility)
+
+  # TODO
+  # Once the new drop_exists is done, one must check to see if a file/folder
+  # exists on Dropbox before proceeding
+
+  path <- add_slashes(path)
+  settings <-
+    drop_compact(
```

Comment:
is this necessary? will `encode = "json"` not handle depth properly?

Reply:
I had that before adding encode. I'll make a note to test this before removing but I think we can move ahead with the pr and I will address this later.